modules/phase_field/include/kernels/AllenCahnElasticEnergyOffDiag.h

```diff
@@ -0,0 +1,52 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef ALLENCAHNELASTICENERGYOFFDIAG_H
+#define ALLENCAHNELASTICENERGYOFFDIAG_H
+
+#include "DerivativeMaterialInterface.h"
+#include "JvarMapInterface.h"
+#include "Kernel.h"
+
+// Forward Declarations
+class AllenCahnElasticEnergyOffDiag;
+class RankTwoTensor;
+
+template <>
+InputParameters validParams<AllenCahnElasticEnergyOffDiag>();
+
+/**
+ * This kernel computes the off-diagonal jacobian of elastic energy in AllenCahn respect to
+ * displacements
+ */
+class AllenCahnElasticEnergyOffDiag
+  : public DerivativeMaterialInterface<JvarMapKernelInterface<Kernel>>
+{
+public:
+  AllenCahnElasticEnergyOffDiag(const InputParameters & parameters);
+
+protected:
+  Real computeQpResidual() override { return 0.0; }
+
+  Real computeQpJacobian() override { return 0.0; }
+
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
+
+  /// Mobility
+  const MaterialProperty<Real> & _L;
```

Comment:
Just a remark: technically symbol names consisting of a leading underscore followed by a single uppercase letter are reserved for internal use of the compiler. We've encountered cases where this breaks the build of MOOSE (on Windows).

We have plenty of those in MOOSE but we really shouldn't add more.